File: springapp/CRMS_Backend/src/main/java/com/examly/springapp/service/UserService.java
package com.examly.springapp.service;

import java.util.List;
import java.util.Optional;

import com.examly.springapp.model.User;

public interface UserService {

    User create(User user);

    List<User> read();

    Optional<User> getById(Long id);

    User update(User user);

    void delete(Long id);
}

File: springapp/CRMS_Frontend/src/services/UserService.js
import api from '../utils/api';

const UserService = {
    getAll: async () => {
        const response = await api.get('/users');
        return response.data;
    },
    getById: async (id) => {
        const response = await api.get(`/users/${id}`);
        return response.data;
    },
    create: async (user) => {
        const response = await api.post('/users', user);
        return response.data;
    },
    update: async (id, user) => {
        const response = await api.put(`/users/${id}`, user);
        return response.data;
    },
    delete: async (id) => {
        await api.delete(`/users/${id}`);
    }
};

export default UserService;
